libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraComponent.kt

```diff
@@ -38,29 +41,17 @@ internal class CameraComponent constructor(
         ),
 ) : UIComponent() {
 
-    // TODO: Remove or improve to calculate the viewport size at runtime
-    private val overviewEdgeInsets = EdgeInsets(
-        Utils.dpToPx(180f).toDouble(),
-        Utils.dpToPx(80f).toDouble(),
-        Utils.dpToPx(180f).toDouble(),
-        Utils.dpToPx(80f).toDouble(),
-    )
-    private val followingEdgeInsets = EdgeInsets(
-        Utils.dpToPx(0f).toDouble(),
-        Utils.dpToPx(0f).toDouble(),
-        Utils.dpToPx(200f).toDouble(),
-        Utils.dpToPx(0f).toDouble(),
-    )
-
     private val gesturesHandler = NavigationBasicGesturesHandler(navigationCamera)
+
     // To determine if [$this] is a fresh instantiation and is garbage collected upon onDetached
     private var isFirstAttached: Boolean = true
+    private val resources get() = mapView.resources
 
     override fun onAttached(mapboxNavigation: MapboxNavigation) {
         super.onAttached(mapboxNavigation)
         mapView.camera.addCameraAnimationsLifecycleListener(gesturesHandler)
-        viewportDataSource.overviewPadding = overviewEdgeInsets
-        viewportDataSource.followingPadding = followingEdgeInsets
+        viewportDataSource.overviewPadding = getOverviewEdgeInsets()
```

Comment:
Each layout will sizes depending on what is inside. Have you been thinking about making the padding update?

Reply:
Wondering if we need to ask the map team to add support for observing "EdgeInsets"

Reply:
Do you mean making the padding dynamic and re-calculate on every `onLayout`? We (me and @abhishek1508) thought about this and decided to to perform viewport padding calculations only once per navigation state change. Since the CameraComponent is always re-bound for every NavigationState doing this does just that.
If we ever change CameraComponent to be persistent for all states we will wrap this with navigationState flow observer.

Reply:
Let's hold onto this PR until we have [this](https://github.com/mapbox/mapbox-navigation-android/pull/5578) and [this](https://github.com/mapbox/navigation-sdks/issues/1528) merged. Some things in this implementation will make more sense once we have these 2 issues addressed.

Reply:
It makes sense. I also opened a issue with maps in case they would be willing to add support for observing the edgeinsets https://github.com/mapbox/mapbox-maps-android/issues/1229